### PR TITLE
feat: align skill frontmatter parser with Agent Skills spec #313

### LINKED ISSUE #313: Align skill frontmatter parser with Agent Skills spec
### QuickApps version

latest

### Area

Skills

### What is the current state?

`quickapp.skills._frontmatter.parse_frontmatter` — shared by predefined skills (`AgentSkillsProvider`) and DIAL prompt skills (`DialPromptSkillResolver`) — is stricter than the cross-client [Agent Skills client-implementation guide](https://agentskills.io/client-implementation/adding-skills-support#step-2-parse-skill-md-files):

- The frontmatter regex demands a trailing newline after the closing `---` and rejects BOM, leading whitespace, and leading blank lines.
- `name` length > 64 chars and non-conforming `name` format raise a hard `SkillValidationError` (spec: warn, load anyway).
- A name that doesn't match the parent directory causes predefined skills to be silently skipped (spec: warn, load anyway).
- YAML parse failures (e.g. unquoted colon in `description`) surface as-is, with no quote / block-scalar retry.

### Why is this debt?

SKILL.md content authored for other clients gets rejected after a round trip through DIAL prompt storage or a web textarea (stripped trailing newline, added BOM, etc.). Error messages like *"No YAML frontmatter found"* are misleading — the frontmatter is present, only the framing differs. The strictness also reduces interoperability with the `.agents/skills/` convention adopted by other clients.

### Proposed remediation

Adopt the spec's "lenient with diagnostics" model:

1. Loosen the frontmatter regex — tolerate optional BOM, leading whitespace / blank line, and a closing fence at end-of-file.
2. Downgrade `name` length, `name` format, and name/parent-dir mismatch from hard errors to logged warnings; load the skill anyway. Missing `name` / `description` and unparseable YAML remain hard failures.
3. Add a YAML quote- / block-scalar retry for the unquoted-colon case before failing.
4. Return the trimmed body from the parser so downstream consumers don't re-strip frontmatter ad-hoc.
5. Update parser unit tests for each relaxed case.

### Alternatives considered

Regex-only fix without relaxing name validation — leaves imported skills with long or oddly-cased names broken and doesn't address the YAML edge cases.

### Additional information

Spec reference: https://agentskills.io/client-implementation/adding-skills-support#step-2-parse-skill-md-files

### PR DESCRIPTION
**Pull request by ypldan**
### Applicable issues

- fixes #313

### Description of changes

Aligns `parse_frontmatter` with the cross-client [Agent Skills client guide](https://agentskills.io/client-implementation/adding-skills-support#step-2-parse-skill-md-files)'s "lenient with diagnostics" model. The parser is the shared entry point for the predefined-skills loader (`AgentSkillsProvider`) and the DIAL prompt resolver, so both paths benefit.

- **Loose frontmatter regex**: tolerates BOM, leading whitespace/blank line, closing fence at end-of-file, and CRLF — the variants that get introduced when SKILL.md content round-trips through DIAL prompt storage or a web textarea.
- **Cosmetic checks → warnings**: oversize `name` / `description` / `compatibility` and non-conforming `name` format are recorded in `ParsedSkill.warnings` and logged; the skill still loads. Predefined skills whose `name` doesn't match the parent directory load instead of being silently skipped.
- **YAML quote-retry**: `description: Use when: the user asks` and similar unquoted-colon authoring mistakes are auto-repaired before the parser gives up.
- **Hard failures kept**: no frontmatter, missing `name` / `description`, and YAML still unparseable after the retry.

No schema change.

### Checklist

- [x] Title of the pull request follows [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] App schema changes are backward compatible, or breaking changes are documented with a migration guide

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/quickapp/application/_initialization_error_handler.py [hunk 1]
```diff
@@ -59,7 +62,8 @@ def handle_initialization_issues(self) -> None:
             elif isinstance(exc, SkillCatastrophicInitializationException):
                 catastrophic_lines.append(f"- {exc.reason}")
             elif isinstance(exc, SkillInitializationException) and exc.url is not None:
-                per_url_lines.append(f"- **{exc.url}**: {exc.reason}")
+                bucket = per_url_warning_lines if exc.severity == "warning" else per_url_error_lines
```
Comment (korotaav48) [inline src/quickapp/application/_initialization_error_handler.py]: The bucket is not used further in the code

**Review by korotaav48 — CHANGES_REQUESTED** [review]

File: src/quickapp/application/_initialization_error_handler.py (hunk 1 above)
Comment (ypldan) [reply src/quickapp/application/_initialization_error_handler.py]: fixed

**Review by korotaav48 — APPROVED** [review]
